genum/internal/simple_enum_test.go

```diff
@@ -143,10 +147,11 @@ func TestMyEnum3(t *testing.T) {
 		{enum: internal.MyEnum3(99), invalid: true},
 	}
 
-	testRunner[internal.MyEnum3](t, tests)
+	testRunner[internal.MyEnum3](t, internal.ParseMyEnum3, tests)
```

Comment:
let's put the parse function back so none of this has to change. It shouldn't be necessary to pass a parse function around when you already know the type.

Reply:
ok i reverted that